### ISSUE TITLE
Extraction in post_process context doesn't appear to handle array inputs

### ISSUE THREAD

**Issue by kstaken** [terascope/teraslice#995]
It appears that `extraction` in a post_process context does not appear to handle an array input in the expected manner

```
{ "selector": "somefield:value", "source_field": "inputarray", "start": "value=", "end": "EOP", "target_field": "output", "tag": "myoutput", "multivalue": true }
```

This rule takes an array as input and outputs an array.

The desire is to then follow that rule with another extraction on the output array and also produce an array as output.

```
{ "follow": "myoutput", "post_process": "extraction", "source_field": "output", "start": "|", "end": "EOP", "target_field": "output_separated", "multivalue": true, "mutate": true }
```

This will work if the input is a single value, but not when the input is an array. The desired outcome would be for the extraction to run against every element in the array and then output all the extractions in an array.

**Comment by kstaken** [comment]
Also somewhat related to this ... does the 'multivalue' logic allow multiple rules to match and assign values into the same array or does it always generate a new array? If the latter that is not desirable.

**Comment by kstaken** [comment]
Something else that is occurring in some scenarios is the generation of a field with what I'm going to guess is an array reference appended like `output_separated0`.

**Comment by kstaken** [comment]
The `output_separated0` appears to happen when you have more than one rule labeled as multivalue. Only the last rule labeled with `multivalue: true` works.

**Comment by kstaken** [comment]
resolved in #1028
